addons/hide-ads/userscript.js

```diff
@@ -0,0 +1,201 @@
+    //These are just made by chatGPT after giving it a lot of example comments
+    const repeatedCharactersRegex = /(.)\1{20,}/;
+    const meaninglessRegex = /^[\W\s]*$/;
+
+    if (repeatedCharactersRegex.test(content)) {
+      return true;
+    }
+
+    if (meaninglessRegex.test(content)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  function handleDisabled(commentContent, profileReplyButton, comment, handleProfileReplyButton, blockType) {
+    commentContent.innerHTML = idToContent[commentContent.closest(".comment").getAttribute("id")];
+    commentContent.classList.remove("advertising");
+    comment.classList.remove("contains-advertising");
+
+    if (handleProfileReplyButton()) {
+      profileReplyButton.style.display = "inline";
+    }
+
+    commentContent.style.cursor = "";
+  }
+
+  function handleReenabled(commentContent, profileReplyButton, comment, handleProfileReplyButton, blockType) {
+    commentContent.innerHTML =
+      blockType === "content" ? advertisingContent : blockType === "user" ? advertisingUser : advertisingSpam;
+    commentContent.classList.add("advertising");
+    comment.classList.add("contains-advertising");
+
+    if (handleProfileReplyButton()) {
+      profileReplyButton.style.display = "";
+    }
+
+    commentContent.style.cursor = "pointer";
+  }
+
+  let settings = {};
+
+  function getNewSettings() {
+    settings["projects"] = addon.settings.get("projects");
+    settings["studios"] = addon.settings.get("studios");
+    settings["profiles"] = addon.settings.get("profiles");
+    settings["method"] = addon.settings.get("method");
+    settings["spamFilter"] = addon.settings.get("spamFilter");
+    settings["matches"] = addon.settings.get("matches");
+  }
+
+  getNewSettings();
+
+  function handleComment(element, blockType) {
+    if (addon.settings.get("method") === "hide") {
+      element.style.display = "none";
+    } else {
+      const comment = element.classList.contains("comment") ? element : element.querySelector(".comment");
+      comment.classList.add("contains-advertising");
+
+      let commentContent;
+      let profileReplyButton;
+
+      if (url.includes(settingsToUrl["profiles"])) {
+        commentContent = element.querySelector(".info>.content");
+        profileReplyButton = commentContent.nextElementSibling.lastElementChild;
+      } else {
+        commentContent = element.querySelector(".comment-content");
+      }
+
+      function handleProfileReplyButton() {
+        if (profileReplyButton?.classList.contains("reply") && url.includes(settingsToUrl["profiles"])) {
+          return true;
+        } else return false;
+      }
+
+      const key = comment.getAttribute("id");
+      const value = commentContent.innerHTML;
+
+      idToContent[key] = value;
+
+      commentContent.innerHTML =
+        blockType === "content" ? advertisingContent : blockType === "user" ? advertisingUser : advertisingSpam;
+      commentContent.classList.add("advertising");
```

Comment:
For accessibility, `role="button"` and `tabindex="0"` would be good:

```suggestion
      commentContent.classList.add("advertising");
      commentContent.role = "button";
      commentContent.tabIndex = 0;
```

Reply:
i dont actually know what this is for, can you explain why this is a thing?

Reply:
`role="button"` makes screen readers know that it's a button, and `tabindex="0"` allows you to navigate to the button with the Tab key.

Reply:
ooh ok thanks!

Reply:
You can set this to resolved.